### PR TITLE
Check error comparability for avoiding panic

### LINKED ISSUE #22: panic occured when wrap go-playground./validator's errors
from v4.0.0

```go
package main

import (
	"fmt"

	"github.com/srvc/fail"
	validator "gopkg.in/go-playground/validator.v9"
)

func main() {
	err := validator.ValidationErrors{}
	fmt.Println(fail.Wrap(err))
}
```

```
panic: runtime error: comparing uncomparable type validator.ValidationErrors

goroutine 1 [running]:
github.com/srvc/fail.extractPkgError(0x1be6a0, 0x40cfe0, 0x450230, 0x277c00)
	/go/src/github.com/srvc/fail/pkgerrors.go:82 +0x960
github.com/srvc/fail.convertPkgError(0x1be6a0, 0x40cfe0, 0x1901c0, 0x9f442304)
	/go/src/github.com/srvc/fail/pkgerrors.go:22 +0x40
github.com/srvc/fail.Unwrap(0x1be6a0, 0x40cfe0, 0x15e901, 0x7622)
	/go/src/github.com/srvc/fail/error.go:113 +0x260
github.com/srvc/fail.Wrap(0x1be6a0, 0x40cfe0, 0x0, 0x0, 0x0, 0x4423a0, 0x0, 0x0)
	/go/src/github.com/srvc/fail/error.go:85 +0x40
main.main()
	/go/src/29f35e98-5c00-4964-a695-ff39d667e9e2/main.go:12 +0xa0
```

### PR DESCRIPTION
**Pull request by izumin5210**
To fix #22

### CODE REVIEW

File: pkgerrors.go [hunk 1]
```diff
@@ -79,8 +80,15 @@ func extractPkgError(err error) *pkgError {
 		break
 	}
 
-	if len(stackTraces) == 0 && rootErr == err {
```
Comment (creasty) [inline pkgerrors.go]: c.f., https://github.com/srvc/fail/pull/19#discussion_r268486441

**Comment by creasty** [comment]
https://github.com/srvc/fail/pull/26/commits/4dc2444760cf27e435c692cb158ffd17ba314a84

```diff
-		} else if reflect.DeepEqual(rootErr, err) {
+		} else {
```

1. `err == nil` ⇒ `rootErr == nil`
1. `err != nil` ⇒ `rootErr != nil` ∧ ( `rootErr == err` ∨ `rootErr != err` )

なのでこの比較は不要。

**Comment by qnighy** [comment]
Any plan to release this?

**Comment by creasty** [comment]
@qnighy Sorry!... Here's the new version:
https://github.com/srvc/fail/releases/tag/v4.1.0

**Pull request merged**